src/Drupal/Driver/Fields/Drupal7/AbstractHandler.php

```diff
@@ -22,8 +26,8 @@
    * @param $field_name
    * @return mixed
    */
-  public function __construct($entity_type, $field_name) {
+  public function __construct($entity_type, $field_name, $language) {
     $this->field_info = field_info_field($field_name);
+    $this->language = field_is_translatable('node', $this->field_info) ? $language : LANGUAGE_NONE;
```

Comment:
Field handlers are actually supposed to work with all type of entities, so any specific entity type reference (i.w. ```'node'``` in this case) should be removed, ```$entity_type``` should be used instead.